backend/app/chat/routers.py

```diff
@@ -35,8 +39,23 @@ async def chat(
 
     message_embeddings = await create_embeddings(chat_request.message)
     similar_chunks = await get_similar_n_chunks(
-        message_embeddings, n_similar=5, asession=asession
+        message_embeddings, n_similar=N_TOP_CONTENT, asession=asession
     )
+    if USE_CROSS_ENCODER == "True" and (N_TOP_RERANK > N_TOP_CONTENT):
+        raise ValueError(
+            (
+                "N_TOP_RERANK should be less than or equal to N_TOP_CONTENT "
+                "when using cross-encoder"
+            )
+        )
+
+    if USE_CROSS_ENCODER == "True" and len(similar_chunks) > 0:
```

Comment:
Nit: there's no need to re-rank if we only have 1 doc surely 😅

Reply:
Good point! I'll make this > 1